core/modules/user/user.admin.inc

```diff
@@ -178,6 +194,11 @@ function user_admin_settings($form, &$form_state) {
     '#default_value' => $config->get('user_email_verification'),
     '#description' => t('New users will be required to validate their e-mail address prior to logging into the site, and will be assigned a system-generated password. With this setting disabled, users will be logged in immediately upon registering, and may select their own passwords during registration.')
   );
+  $form['registration_cancellation']['user_email_match'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('When an email address is used as username, require a matching email address.'),
+    '#default_value' => $config->get('user_email_match'),
+  );
```

Comment:
After reviewing this entire patch, I'd like to know if we could nix this entire option. It seems obvious that you shouldn't be allowed to use a different e-mail address as your name than your real e-mail address. This option just muddles the UI.